### PR TITLE
Claim_to

### PR DESCRIPTION
**Pull request by MSghais**
This draft PR doesn't change the old claim function and content request.

Adding a new one for claim_to with the specific content.

Claim_to: function with Starknet recipient in Nostr event content.
ClaimContent: for content request used by claim_to(SocialRequest<ClaimContent>)

### CODE REVIEW

File: onchain/src/social/deposit.cairo [hunk 1]
```diff
@@ -1,20 +1,42 @@
...
 use super::request::{SocialRequest, SocialRequestImpl, SocialRequestTrait, Encode, Signature};
 
 pub type DepositId = felt252;
 
+// TODO add starknet_recipient as a Contract address
+// Find a way to format ContractAddress
+#[derive(Clone, Debug, Drop, Serde)]
+pub struct ClaimContent {
+    pub deposit_id: DepositId,
+    pub starknet_recipient: felt252
```
Comment (maciejka) [inline onchain/src/social/deposit.cairo:13]: It should be a `ContractAddress`

File: onchain/src/social/deposit.cairo [hunk 2]
```diff
@@ -1,20 +1,42 @@
...
         @format!("claim {}", self)
     }
 }
 
+// TODO: Find a way to format ContractAddress
+// Implement Display trait for Contract address
+impl ClaimContentEncodeImpl of Encode<ClaimContent> {
+    fn encode(self: @ClaimContent) -> @ByteArray {
+        // let felt_recipient:felt252=self.starknet_recipient.format_as_byte_array(16);
+        @format!("claim {} to {:?}", self.deposit_id, self.starknet_recipient)
```
Comment (maciejka) [inline onchain/src/social/deposit.cairo:29]: Why `{:?}`?

File: onchain/src/social/deposit.cairo [hunk 3]
```diff
@@ -252,6 +276,39 @@ pub mod DepositEscrow {
                     }
                 );
         }
+
+
+        fn claim_to(ref self: ContractState, request: SocialRequest<ClaimContent>,// starknet_recipient:ContractAddress // Maybe not needed
```
Comment (maciejka) [inline onchain/src/social/deposit.cairo:281]: `starknet_recipient` is not necessary here

**Review by maciejka — CHANGES_REQUESTED** [review]

File: onchain/src/social/deposit.cairo (hunk 2 above)
Comment (MSghais) [reply onchain/src/social/deposit.cairo:29]: Without the {:?}, the ContractAddress can't be format and display. 
We need maybe to do a Display Format trait for ContractAddress?

**Review by maciejka — APPROVED** [review]

**Pull request merged**